### PR TITLE
Fix /simplify Neuron archdir detection

### PR DESCRIPTION
**Pull request by ferdonline**
Instead of relying on build system dependent, we use (parse)
nrnivmodl to get the exact same
This was affecting mac users, as well as the FMT lib

### CODE REVIEW

File: var/spack/repos/builtin/packages/neuron/package.py [hunk 1]
```diff
@@ -368,23 +368,10 @@ def archdir(self):
...
-                    os.path.basename(
-                        os.path.dirname(os.path.dirname(file_list[0]))
-                    )
-            else:
-                neuron_arch = ""
-        return neuron_arch
+        return subprocess.check_output(
+            ['awk', '-F=', '$1 == "MODSUBDIR" { print $2; exit; }',
+             str(self.prefix.bin.nrnivmodl)]
+        ).decode().strip()
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/neuron/package.py]: That's a lot simpler :)

**Review by matz-e — DISMISSED** [review]

**Review by pramodk — DISMISSED** [review]

**Review by iomaganaris — DISMISSED** [review]

**Comment by ferdonline** [comment]
CI says: `python_version.py::test_package_module_compatibility FAILED : AssertionError: assert not {(2, 7): `
Do we really want to force py27 compat? 😞

**Comment by pramodk** [comment]
> Do we really want to force py27 compat? 😞

I would put in other words - I wouldn't break it if there is no major issue. The reason being, if we start building stack on system where there is only Python2 on login nodes (e.g. Marconi), we can build everything ourselves from scratch.

**Comment by matz-e** [comment]
I think ditching Python 2 is a real option with future Spack

**Comment by pramodk** [comment]
> I think ditching Python 2 is a real option with future Spack

In that case we assume that Python3 is **always** available where we want to build the software stack. I would defer this switch to Python3.

**Comment by matz-e** [comment]
> > I think ditching Python 2 is a real option with future Spack
> 
> In that case we assume that Python3 is **always** available where we want to build the software stack. I would defer this switch to Python3.

This is a discussion upstream. Spack will already pick up an installed Python 3 and use it, I guess it's just a question of time until support for 2 is dropped.

If I dig out my old copy of Ubuntu Warty Warthog, Spack will not run there, either…

**Comment by pramodk** [comment]
> This is a discussion upstream.

Exactly! (and IIRC discussions, this will be a conservative choice)

If you upgrade from `Warty Warthog` to `Intrepid Ibex` it should work!

**Comment by ferdonline** [comment]
Ohh, it's actually worse than that. It works under 2.7. Oh gosh, 2.6 is like yet another language...
```
---------------------------- Captured stdout call -----------------------------
==> Spack must remain compatible with Python version 2.6
==> These files require version 2.7
```

**Comment by ferdonline** [comment]
My solution might seem a bit harsh, but I'm pretty sure we have recipes that need at least py2.7 to run, so to stop this 2.6 ideology, can we allow our recipes to use py2.7? @pramodk ?

File: lib/spack/spack/test/python_version.py [hunk 2]
```diff
@@ -89,14 +89,15 @@ def pyfiles(search_paths, exclude=()):
                     yield os.path.join(root, filename)
 
 
-def check_python_versions(files):
+def check_python_versions(files, warn_only_py26=False):
```
Comment (matz-e) [inline lib/spack/spack/test/python_version.py]: Mucking around in core will make future merges harder. I'm actually trying to move away from all the workarounds we have in there.

**Review by matz-e — CHANGES_REQUESTED** [review]

File: lib/spack/spack/test/python_version.py (hunk 2 above)
Comment (ferdonline) [reply lib/spack/spack/test/python_version.py]: In the next change they do to this file I do hope they get rid of python 2.6 compatibility enforcement. Do you suggest some lighter changes?

Comment (matz-e) [reply lib/spack/spack/test/python_version.py]: Revert the `lib` changes.

File: var/spack/repos/builtin/packages/neuron/package.py (hunk 1 above)
Comment (matz-e) [inline var/spack/repos/builtin/packages/neuron/package.py]: It's not much more verbose:
```suggestion
        return subprocess.Popen(
            ['awk', '-F=', '$1 == "MODSUBDIR" { print $2; exit; }',
             str(self.prefix.bin.nrnivmodl)],
             stdout=subprocess.PIPE
        ).communicate()[0].decode().strip()
```

**Review by matz-e — CHANGES_REQUESTED** [review]

**Review by matz-e — APPROVED** [review]
Thanks!

**Comment by ferdonline** [comment]
Tests failed because of a glitch in the testing. I have created a fix in https://bbpcode.epfl.ch/code/#/c/50860/

**Pull request merged**